### PR TITLE
Issue797 add co2 concentration to rom

### LINKED ISSUE #797: Add CO2 Concentration to Reduced Order Model
**What is the problem?**

Our Project is about adding a CO2 concentration to the ReducedOrder Model.

**How do we want to solve it? Describe the solution you'd like**

We work with trace substances to calculate the CO2 concentration over time in the zone.
We want to add a connector use_C_flow to Indoor Air to calculate the CO2 concentration.

**Describe alternatives you've considered**

We first tried to work with the fluid ports but we decided to work with real connectors to keep it simple.

### PR DESCRIPTION
**Pull request by katharinabr**
This closes #797.

Implementation of co2 balance for rom.
Update of fluid model to consider trace substances in thermal zone and multizone.
Manual merge of ibpsa changes to oneelement.

### CODE REVIEW

File: AixLib/BoundaryConditions/InternalGains/CO2/CO2Balance.mo [hunk 1]
```diff
@@ -0,0 +1,130 @@
+within AixLib.BoundaryConditions.InternalGains.CO2;
+model CO2Balance "Calculation of CO2 concentration within a thermal zone"
+  parameter Modelica.SIunits.Area AreaZon "Zone area";
+  parameter Real actDeg = 1.8 "Activity degree (Met units)";
+  parameter Modelica.SIunits.Volume VZon "Zone volume";
+  parameter Modelica.SIunits.MassFraction XCO2_amb=6.12157E-4
+    "Massfraction of CO2 in atmosphere (equals 403ppm)";
+  parameter Modelica.SIunits.Area AreaBod=1.8
```
Comment (KremerMartin) [inline AixLib/BoundaryConditions/InternalGains/CO2/CO2Balance.mo:8]: should be lower case: areaBod

File: AixLib/BoundaryConditions/InternalGains/CO2/CO2Balance.mo [hunk 2]
```diff
@@ -0,0 +1,130 @@
+within AixLib.BoundaryConditions.InternalGains.CO2;
+model CO2Balance "Calculation of CO2 concentration within a thermal zone"
+  parameter Modelica.SIunits.Area AreaZon "Zone area";
```
Comment (KremerMartin) [inline AixLib/BoundaryConditions/InternalGains/CO2/CO2Balance.mo:3]: should be lower case

File: AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+        package Medium =
+          AixLib.Media.Air (                                                                                            extraPropertiesNames={"C_flow"})));
+
+  // CO2 parameters
+  parameter Real actDeg=1.8 "Activity degree (Met units)"
+    annotation (Dialog(tab="CO2", enable=use_CO2_balance));
+  parameter Modelica.SIunits.MassFraction XCO2_amb=6.12157E-4
+    "Massfraction of CO2 in atmosphere (equals 403ppm)"
+    annotation (Dialog(tab="CO2", enable=use_CO2_balance));
+  parameter Modelica.SIunits.Area AreaBod=1.8
```
Comment (KremerMartin) [inline AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo:14]: lower case

File: AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo [hunk 4]
```diff
@@ -49,6 +64,21 @@ model ThermalZoneMoistAirExchange
     "Ventilation and infiltration humidity" annotation (Placement(
         transformation(extent={{-120,-90},{-80,-50}}), iconTransformation(
           extent={{-126,-80},{-100,-54}})));
+  BoundaryConditions.InternalGains.CO2.CO2Balance cO2Balance(
+    AreaZon=zoneParam.AZone,
```
Comment (KremerMartin) [inline AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo:68]: lower case

File: AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo [hunk 5]
```diff
@@ -49,6 +64,21 @@ model ThermalZoneMoistAirExchange
...
+    actDeg=actDeg,
+    VZon=zoneParam.VAir,
+    XCO2_amb=XCO2_amb,
+    AreaBod=AreaBod,
```
Comment (KremerMartin) [inline AixLib/ThermalZones/ReducedOrder/ThermalZone/ThermalZoneMoistCO2AirExchange.mo:72]: lower case

**Review by KremerMartin — CHANGES_REQUESTED** [review]
For successful check, the CO2-models have to be made conditional with the same condition as the ROM-model.
Some upper case parameters have to be changed to lower case.

**Comment by MartinRaetz** [comment]
As @martinkremer made a thorough code check, I may test this implementation for its function in a use case. 
If I understood correctly, it should be usable besides the failing check?
(I will tackle this approximately next week)

**Review by KremerMartin — APPROVED** [review]

**Comment by KremerMartin** [comment]
@katharinabr @MartinRaetz I have approved the changes. (Sorry, that it took me so long.) The ci is failing, as the checks for the thermal zone fail. I have no clue about that, because in dymola all checks are successful.

**Review by MartinRaetz — APPROVED** [review]

**Pull request merged**